lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1819,7 +1821,7 @@ void InferenceMapping::constrainTypes(Value larger, Value smaller) {
         } else if (type.isGround()) {
           // Leaf element, look up their expressions, and create the constraint.
           constrainTypes(getExpr(FieldRef(larger, fieldID)),
-                         getExpr(FieldRef(smaller, fieldID)));
+                         getExpr(FieldRef(smaller, fieldID)), false, equal);
```

Comment:
Don't love this re:bools and ergonomics but wanted to get this reviewed for functionality first, but once that seems good suggestions on improvements welcome (please!). FWIW--  imposeUpperBounds adds LEQ in compatible way to GEQ (if x >= y is requested, it'll add y <= x if x isn't free variable), where-as "equal" causes LEQ (x >= y && x <= y) to be added.  Anyway.